etna/articles/models.py

```diff
@@ -269,48 +271,67 @@ def similar_items(self) -> Tuple["ArticlePage"]:
             return ()
 
         # Identify 'other' live pages with tags in common
-        tag_match_ids = (
-            ArticlePage.objects.public()
-            .live()
-            .not_page(self)
-            .filter(tagged_items__tag_id__in=tag_ids)
-            .values_list("id", flat=True)
-            .distinct()
-        )
-        if not tag_match_ids:
+        matching_page_ids = []
+
+        for page_type in [ArticlePage, FocusedArticlePage, RecordArticlePage]:
+            matching_page_ids.extend(
+                page_type.objects.live()
+                .public()
+                .not_page(self)
+                .filter(tagged_items__tag_id__in=tag_ids)
+                .values_list("id", flat=True)
+                .distinct()
+            )
+
+        if not matching_page_ids:
             # Avoid unncecssary lookups
             return ()
 
         # Use search() to prioritise items with the highest number of matches
-        return tuple(
-            ArticlePage.objects.filter(id__in=tag_match_ids).search(
-                self.article_tag_names,
-                fields=["article_tag_names"],
-                operator="or",
-            )[:3]
-        )
+        relevant_pages = []
+
+        for page_type in [ArticlePage, FocusedArticlePage, RecordArticlePage]:
+            relevant_pages.extend(
+                page_type.objects.filter(id__in=matching_page_ids).search(
+                    self.article_tag_names,
+                    fields=["article_tag_names"],
+                    operator="or",
+                )
+            )
+
+        return tuple(relevant_pages[:3])
```

Comment:
Similar thing here... unless we're using `select_related` and `prefetch_related` to fetch additional things we need, we're just tripling the work here, and not gaining anything from a performance perspective.

```suggestion
        return tuple(
            Page.objects.filter(id__in=matching_page_ids).search(
                self.article_tag_names,
                fields=["article_tag_names"],
                operator="or",
            )[:3]
        )
```

Reply:
Hey Andy, not sure if you're still replying on here but I can't use `.search(self.article_tag_names)` on `Page` as it doesn't have `article_tag_names` as a `search_field` :(

Reply:
Ah, in that case, I think we should go back to the drawing board on this one. It was a stretch to use one search for this, but three is definitely too much. Maybe we can work out the position in python, then do a sort based on that?

Reply:
Just be sure to use `prefetch_related` to fetch the tags for all matches in one go 😜

Reply:
Do we absolutely need `search`? Is it for the level of relevance against all of the tags?

Reply:
I've updated to use search but currently it's searching the tags against every field, not the tags on the other pages

Reply:
I'm not sure if searching on a queryset of `Page` objects will match on the custom fields - hence the recommendation. Equally viable is forget about relevancy and just order by `-first_published_at` to prioritise newer content.

Reply:
Hey @ababic, I tried implementing the `|` in the type hinting but have had no luck because "ArticlePage" isn't defined.

I've fixed up the tests, and everything else - I know you're off the project now though so if you'd like me to move this on to someone else, then I can :)

Cheers! 😃